lua/nvim-tree.lua

```diff
@@ -363,15 +363,15 @@ local function setup_autocommands(opts)
 
   create_nvim_tree_autocmd("BufReadPost", {
     callback = function()
-      if filters.config.filter_no_buffer or renderer.config.highlight_opened_files then
+      if filters.config.filter_no_buffer or renderer.config.highlight_opened_files ~= "none" then
         reloaders.reload_explorer()
       end
     end,
   })
 
   create_nvim_tree_autocmd("BufUnload", {
     callback = function(data)
-      if filters.config.filter_no_buffer or renderer.config.highlight_opened_files then
+      if filters.config.filter_no_buffer or renderer.config.highlight_opened_files ~= "none" then
```

Comment:
```suggestion
        utils.debounce("BufUnload:filter_buffer", 10, function()
          reloaders.reload_explorer(nil, data.buf)
        end)
```
Same as above

Reply:
changes made, tested and works as expected. please review again :smile: